test/optional-test.js

```diff
@@ -311,5 +311,64 @@ describe('Optional.js', function () {
             // Here to complete the Java Optional API. Completely useless.
             assert.strictEqual(nonNullOptional.hashCode(), -1);
         });
+
+        it('.ifPresentOrElse() if value is present calls the consumer and not the else callable', function () {
+            var consumerCalled = false,
+                elseCallableCalled = false,
+                passedValue;
+
+            nonNullOptional.ifPresentOrElse(
+                function consumer(value) {
+                    consumerCalled = true;
```

Comment:
I should probably add Sinon & Mocha - a decent stubbing lib would make these tests much less hassle to write